enhancements/network/bgp-ovn-kubernetes.md

```diff
@@ -0,0 +1,701 @@
+external, physical network of the cluster which a user administers will be called the “provider network”.
+
+### Importing Routes from the Provider Network
+
+Today in OpenShift there is no API for a user to be able to configure routes into OVN. In order for a user to change how
+cluster traffic is routed egress into the cluster, the user leverages local gateway mode, which forces egress traffic to
+hop through the Linux host networking stack. There a user can configure routes inside the host via NM State. This
+manual configuration would need to be performed and maintained across nodes and VRFs within each node.
+
+Additionally, if a user chooses to not manage routes within the host for local gateway mode, or the user chooses shared
+gateway mode, then by default traffic is always sent to the default gateway. The only other way to affect egress routing
+is by using the Multiple External Gateways (MEG) feature. With this feature the user may choose to have multiple
+different egress gateways per namespace to send traffic to.
+
+As an alternative, configuring BGP peers and which route-targets to import would eliminate the need to manually
+configure routes in the host, and would allow dynamic routing updates based on changes in the provider’s network.
+
+### Exporting Routes into the Provider Network
+
+There exists a need for provider networks to learn routes directly to services and pods today in Kubernetes. More
+specifically, MetalLB is already one solution where load balancer IPs are advertised by BGP to provider networks. The
+goal of this RFE is to not duplicate or replace the function of MetalLB. MetalLB should be able to interoperate with
+OVN-Kubernetes, and be responsible for advertising services to a provider’s network.
+
+However, there is an alternative need to advertise pod IPs on the provider network. One use case is integration with 3rd
+party load balancers, where they terminate a load balancer and then send packets directly to OCP nodes with the
+destination IP address being the pod IP itself. Today these load balancers rely on custom operators to detect which node
+a pod is scheduled to and then add routes into its load balancer to send the packet to the right node.
+
+By integrating BGP and advertising the pod subnets/addresses directly on the provider network, load balancers and other
+entities on the network would be able to reach the pod IPs directly.
+
+### Datapath Performance
+
+In cases where throughput is a priority, using the underlay directly can eliminate the need for tunnel encapsulation,
+and thus reducing the overhead and byte size of each packet. This allows for greater throughput.
+
+### Multi-homing, Link Redundancy, Fast Convergence
+
+BGP can use multi-homing with ECMP routing in order to provide layer 3 failover. When a link goes down, BGP can reroute
+via a different path. This functionality can be coupled with BFD in order to provide fast failover.
```

Comment:
Failover/redundancy of _what_ exactly?

Reply:
Yes, I understand what the feature means _in general_, I'm asking for the specifics of why you brought it up here since you don't talk about multiple paths or failover anywhere else in the enhancement.

eg, are you saying, perhaps, that you are going to advertise that all pod IPs are reachable via all nodes?

Reply:
No we are not advertising all IPs reachable by any node. The multiple paths is coming from multiple NICs on the host connected to BGP fabric. This is common in spine and leaf topologies. This way if one link goes down to a BGP router, there is a 2nd link available to another leaf, and the failover happens really quickly with BFD. I'll clarify the use case.